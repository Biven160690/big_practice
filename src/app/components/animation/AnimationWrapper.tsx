'use client';

import React from 'react';
import { useSprings } from 'react-spring';

import { styles, getAnimationDelays } from './helper';

export type Props = {
    children: JSX.Element[];
};

export const AnimationWrapper = ({ children }: Props) => {
    const [props] = useSprings(styles.length, (index) => ({
        ...styles[index],
        delay: getAnimationDelays(styles.length)[index],
    }));

    return (
        <React.Fragment>
            {React.Children.map(children, (child, index) =>
                React.cloneElement(child, {
                    style: props[index],
                })
            )}
        </React.Fragment>
    );
};
