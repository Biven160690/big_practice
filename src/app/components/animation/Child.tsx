'use client';

import { animated, useSpring } from 'react-spring';

import styles from './animation.module.scss';

import { commonStyles } from './helper';

export type Props = {
    delay: number;
};

export const Child = ({ delay }: Props) => {
    const style = useSpring({
        ...commonStyles,
        delay,
    });

    return <animated.div className={styles.child} style={style} data-testId="child"/>;
};