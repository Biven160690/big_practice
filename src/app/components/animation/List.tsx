import { animated } from 'react-spring';

import { AnimationWrapper } from './AnimationWrapper';

import styles from './animation.module.scss';

export const List = () => {
    return (
        <div className={styles.awardContainer} data-testid='list'>
            <AnimationWrapper>
                <animated.div className={styles.square} />
                <animated.div className={styles.circle} />
                <animated.div className={styles.triangleTop} />
                <animated.div className={styles.triangleBottom} />
            </AnimationWrapper>
        </div>
    );
};
