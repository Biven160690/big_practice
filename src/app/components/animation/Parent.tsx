'use client';

import { animated, useSpring } from 'react-spring';

import { Child } from './Child';

import { childrenDelay, commonStyles } from './helper';

import styles from './animation.module.scss';

export type Props = {
    elements: number[];
    delay: number;
};

export const Parent = ({ elements, delay }: Props) => {
    const style = useSpring({
        ...commonStyles,
        delay,
    });

    return (
        <animated.div className={styles.parent} style={style} data-testId="parent">
            {elements.map((child, index) => {
                return <Child key={child} delay={delay + childrenDelay * (index + 1)} />;
            })}
        </animated.div>
    );
};