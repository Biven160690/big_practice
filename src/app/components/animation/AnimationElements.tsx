'use client';

import { List } from './List';
import { Parent } from './Parent';

import { example, getAnimationDelays } from './helper';

import styles from './animation.module.scss';

export const AnimationElements = () => {
    return (
        <div className={styles.container}>
            <List />
            {example.map(({ elements }, index) => {
                const delay = getAnimationDelays(
                    example.length,
                    elements.length
                )[index];
                return <Parent key={index} elements={elements} delay={delay} />;
            })}
        </div>
    );
};
