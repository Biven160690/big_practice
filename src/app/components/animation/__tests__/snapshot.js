import { render } from '@testing-library/react';
import { Child } from '../Child';
import { Parent } from '../Parent';
import { AnimationElements } from '../AnimationElements';
import { List } from '../List';
import { AnimationWrapper } from '../AnimationWrapper';

it('renders Child', () => {
    const { container } = render(<Child delay={10} />);
    expect(container).toMatchSnapshot();
});

it('renders Parent', () => {
    const { container } = render(<Parent elements={[1, 2, 3]} delay={10} />);
    expect(container).toMatchSnapshot();
});

it('renders AnimationElements', () => {
    const { container } = render(<AnimationElements />);
    expect(container).toMatchSnapshot();
});

it('renders List', () => {
    const { container } = render(<List />);
    expect(container).toMatchSnapshot();
});

it('renders AnimationWrapper', () => {
    const mockChildren = [
        <div key="1">First child</div>,
        <div key="2">Second child</div>,
    ];
    // eslint-disable-next-line react/no-children-prop
    const { container } = render(<AnimationWrapper children={mockChildren} />);
    expect(container).toMatchSnapshot();
});
