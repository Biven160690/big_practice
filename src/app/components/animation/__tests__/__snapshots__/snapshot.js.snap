// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AnimationElements 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="awardContainer"
      data-testid="list"
    >
      <div
        class="square"
        style="opacity: 0; transform: translateY(-100%);"
      />
      <div
        class="circle"
        style="opacity: 0; transform: translateY(100%);"
      />
      <div
        class="triangleTop"
        style="opacity: 0; transform: translateX(-100%);"
      />
      <div
        class="triangleBottom"
        style="opacity: 0; transform: translateX(100%);"
      />
    </div>
    <div
      class="parent"
      data-testid="parent"
      style="opacity: 0; transform: translateY(-100%);"
    >
      <div
        class="child"
        data-testid="child"
        style="opacity: 0; transform: translateY(-100%);"
      />
      <div
        class="child"
        data-testid="child"
        style="opacity: 0; transform: translateY(-100%);"
      />
      <div
        class="child"
        data-testid="child"
        style="opacity: 0; transform: translateY(-100%);"
      />
      <div
        class="child"
        data-testid="child"
        style="opacity: 0; transform: translateY(-100%);"
      />
      <div
        class="child"
        data-testid="child"
        style="opacity: 0; transform: translateY(-100%);"
      />
      <div
        class="child"
        data-testid="child"
        style="opacity: 0; transform: translateY(-100%);"
      />
    </div>
    <div
      class="parent"
      data-testid="parent"
      style="opacity: 0; transform: translateY(-100%);"
    >
      <div
        class="child"
        data-testid="child"
        style="opacity: 0; transform: translateY(-100%);"
      />
      <div
        class="child"
        data-testid="child"
        style="opacity: 0; transform: translateY(-100%);"
      />
      <div
        class="child"
        data-testid="child"
        style="opacity: 0; transform: translateY(-100%);"
      />
      <div
        class="child"
        data-testid="child"
        style="opacity: 0; transform: translateY(-100%);"
      />
      <div
        class="child"
        data-testid="child"
        style="opacity: 0; transform: translateY(-100%);"
      />
      <div
        class="child"
        data-testid="child"
        style="opacity: 0; transform: translateY(-100%);"
      />
    </div>
  </div>
</div>
`;

exports[`renders AnimationWrapper 1`] = `
<div>
  <div
    style="transform: [object Object];"
  >
    First child
  </div>
  <div
    style="transform: [object Object];"
  >
    Second child
  </div>
</div>
`;

exports[`renders Child 1`] = `
<div>
  <div
    class="child"
    data-testid="child"
    style="opacity: 0; transform: translateY(-100%);"
  />
</div>
`;

exports[`renders List 1`] = `
<div>
  <div
    class="awardContainer"
    data-testid="list"
  >
    <div
      class="square"
      style="opacity: 0; transform: translateY(-100%);"
    />
    <div
      class="circle"
      style="opacity: 0; transform: translateY(100%);"
    />
    <div
      class="triangleTop"
      style="opacity: 0; transform: translateX(-100%);"
    />
    <div
      class="triangleBottom"
      style="opacity: 0; transform: translateX(100%);"
    />
  </div>
</div>
`;

exports[`renders Parent 1`] = `
<div>
  <div
    class="parent"
    data-testid="parent"
    style="opacity: 0; transform: translateY(-100%);"
  >
    <div
      class="child"
      data-testid="child"
      style="opacity: 0; transform: translateY(-100%);"
    />
    <div
      class="child"
      data-testid="child"
      style="opacity: 0; transform: translateY(-100%);"
    />
    <div
      class="child"
      data-testid="child"
      style="opacity: 0; transform: translateY(-100%);"
    />
  </div>
</div>
`;
